Sources/PackageCollectionsSigning/Certificate/CertificatePolicy.swift

```diff
@@ -18,9 +30,362 @@ protocol CertificatePolicy {
+        #else
+        let eku = CCryptoBoringSSL_X509_get_extended_key_usage(certificate.underlying)
+        return eku & UInt32(usage.flag) > 0
+        #endif
+    }
+
+    /// Checks that the certificate supports OCSP. This **must** be done before calling `verify` to ensure
+    /// the necessary properties are in place to trigger revocation check.
+    func supportsOCSP(certificate: Certificate) throws -> Bool {
+        #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+        // Check that certificate has "Certificate Authority Information Access" extension and includes OCSP as access method.
+        // The actual revocation check will be done by the Security framework in `verify`.
+        guard let dict = SecCertificateCopyValues(certificate.underlying, [kSecOIDAuthorityInfoAccess] as CFArray, nil) as? [CFString: Any] else { // ignore error
+            throw CertificatePolicyError.extensionFailure
+        }
+        guard let infoAccessDict = dict[kSecOIDAuthorityInfoAccess] as? [CFString: Any],
+            let infoAccessValue = infoAccessDict[kSecPropertyKeyValue] as? [[CFString: Any]] else {
+            return false
+        }
+        return infoAccessValue.first(where: { valueDict in valueDict[kSecPropertyKeyValue] as? String == "1.3.6.1.5.5.7.48.1" }) != nil
+        #else
+        // Check that there is at least one OCSP responder URL, in which case OCSP check will take place in `verify`.
+        let ocspURLs = CCryptoBoringSSL_X509_get1_ocsp(certificate.underlying)
+        defer { CCryptoBoringSSL_sk_OPENSSL_STRING_free(ocspURLs) }
+
+        return CCryptoBoringSSL_sk_OPENSSL_STRING_num(ocspURLs) > 0
+        #endif
+    }
+}
+
+enum CertificateExtendedKeyUsage {
+    case codeSigning
+
+    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    var data: Data {
+        switch self {
+        case .codeSigning:
+            // https://stackoverflow.com/questions/49489591/how-to-extract-or-compare-ksecpropertykeyvalue-from-seccertificate
+            // https://github.com/google/der-ascii/blob/cd91cb85bb0d71e4611856e4f76f5110609d7e42/cmd/der2ascii/oid_names.go#L100
+            return Data([0x2B, 0x06, 0x01, 0x05, 0x05, 0x07, 0x03, 0x03])
+        }
+    }
+
+    #else
+    var flag: CInt {
+        switch self {
+        case .codeSigning:
+            // https://www.openssl.org/docs/man1.1.0/man3/X509_get_extension_flags.html
+            return XKU_CODE_SIGN
+        }
     }
+    #endif
+}
+
+extension CertificatePolicy {
+    static func loadCerts(at directory: URL) -> [Certificate] {
+        var certs = [Certificate]()
+        if let enumerator = FileManager.default.enumerator(at: directory, includingPropertiesForKeys: nil) {
+            for case let fileURL as URL in enumerator {
+                do {
+                    certs.append(try Certificate(derEncoded: Data(contentsOf: fileURL)))
+                } catch {
+                    // Skip cert if problematic
```

Comment:
we may want to inject (late PR) the diagnostics engine here so we can emit warnings

Reply:
wired it up